.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: test
+
+on:
+  push:
+    branches:
+      - "*"
+  pull_request:
+    branches: [main]
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    container: golang:1.15
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Download Dependencies
+        run: go mod tidy
+
+      - name: CodeCov
+        run: |
+          bash -c '/bin/bash <(curl -s https://codecov.io/bash)'
+
+#     - name: Build
```

Comment:
this should be uncommented and moved before the codecov action.  can't get test coverage if we don't run tests

Reply:
can't run tests, if we don't have tests 😆 
i'll try to write a couple of tests here at the end of the day if i have time

Reply:
I think I'll draft or close this PR until we're ready to actually run some tests.